sdk/core/core-client-rest/test/urlHelpers.spec.ts

```diff
@@ -131,7 +131,7 @@ describe("urlHelpers", () => {
     const result = buildRequestUrl(mockBaseUrl, "/foo", [], {
       queryParameters: { foo: " aaaa", bar: "b= " },
     });
-    assert.equal(result, `https://example.org/foo?foo=+aaaa&bar=b%3D+`);
+    assert.equal(result, `https://example.org/foo?foo=%20aaaa&bar=b%3D%20`);
```

Comment:
This change is a result of using `encodeURIComponent` instead of `URLSearchParams` to encode. We could continue using `URLSearchParams` like I had here: 9510ce333f8bc414c1075d936fbe9d6faa0b7e1e, but @xirzec mentioned to me that there might be issues with spaces being encoded to `+` -- apparently we needed to avoid this behavior for some reason in the old core-client. Will investigate further.

Reply:
I accept all of the blame if encodeURIComponent bites us later, but I've found it is the most likely to be compatible with how our services expect parameters to be encoded.